### PR TITLE
Prm 408 Add agent answers on CYA page

### PR DESCRIPTION
**Pull request by ruthdas03**

### CODE REVIEW

File: app/helpers/SessionAnswersHelper.scala [hunk 1]
```diff
@@ -209,4 +211,39 @@ object SessionAnswersHelper extends ImplicitDateFormatter {
...
+    val agentSession =request.session.get(SessionKeys.agentSessionVrn).isDefined
+
+    if(isAllAnswerPresentForReasonableExcuse(reasonableExcuse) && !agentSession) {
+      getContentForReasonableExcuseCheckYourAnswersPage(reasonableExcuse)
+    }
+      else if(agentSession){
+      getContentForAgentsCheckYourAnswersPage() ++ getContentForReasonableExcuseCheckYourAnswersPage(reasonableExcuse)
+    }
+    else {
+      logger.error(s"[SessionAnswersHelper] User hasn't got all keys in session for reasonable excuses :$reasonableExcuse")
```
Comment (LewisMitchell) [inline app/helpers/SessionAnswersHelper.scala:244]: I don't think you need this extra logging as (in the controller) it is already logged. So I think you can just return the empty Seq.

File: test/helpers/SessionAnswersHelperSpec.scala [hunk 2]
```diff
@@ -663,4 +663,99 @@ class SessionAnswersHelperSpec extends SpecBase {
...
     }
   }
+
+  "getContentForAgentsCheckYourAnswersPage" should {
+    "when the client planned to submit VAT return (so no cause Of LateSubmission chosen)" should {
+      "return a Seq[String, String, String] of answers" in {
+        val fakeRequestWithClientPresent = fakeRequest
+          .withSession(
+            SessionKeys.whoPlannedToSubmitVATReturn -> "client",
+            SessionKeys.causeOfLateSubmissionAgent -> "client"
```
Comment (LewisMitchell) [inline test/helpers/SessionAnswersHelperSpec.scala:673]: You wouldn't need to have `SessionKeys.causeOfLateSubmissionAgent` as that page isn't shown when the user select that the client had an issue.

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor changes 😄

File: app/helpers/SessionAnswersHelper.scala (hunk 1 above)
Comment (ruthdas03) [reply app/helpers/SessionAnswersHelper.scala]: Done

File: test/helpers/SessionAnswersHelperSpec.scala (hunk 2 above)
Comment (ruthdas03) [reply test/helpers/SessionAnswersHelperSpec.scala]: Done

**Review by LewisMitchell — APPROVED** [review]
LGTM.

**Pull request merged**